code/modules/atmospherics/machinery/components/gas_recipe_machines/crystallizer.dm

```diff
@@ -39,6 +39,19 @@
 /obj/machinery/atmospherics/components/binary/crystallizer/Initialize(mapload)
 	. = ..()
 	internal = new
+	register_context()
+
+/obj/machinery/atmospherics/components/binary/crystallizer/add_context(atom/source, list/context, obj/item/held_item, mob/user)
+	. = ..()
+	if(!held_item)
+		return CONTEXTUAL_SCREENTIP_SET
+	context[SCREENTIP_CONTEXT_RMB] = "Turn [on ? "off" : "on"]"
```

Comment:
shouldn't this be above the early return

Reply:
No because to turn the crystallizer off you need to attackby_secondary()
you can't turn it off with an empty hand

Reply:
alllright, that's kinda silly, but alllright

Reply:
@LemonInTheDark you're right i just changed it rn, better to be consistent